addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -455,10 +455,11 @@ function factory(dependencies) {
                 channel.members.includes(this.env.messaging.currentPartner)
             );
 
-            channel = this.env.models['mail.thread'].insert(Object.assign(
-                { isPinned: true },
-                convertedData,
-            ));
+            if (channel) {
+                channel.update(convertedData);
+            } else {
+                channel = this.env.models['mail.thread'].insert(convertedData);
+            }
```

Comment:
Unless I'm missing something this is similar to just doing insert, no need for the condition.